erpnext/setup/doctype/company/company.py

```diff
@@ -223,102 +223,65 @@ def create_default_accounts(self):
 			},
 			{
-				"doctype": "Department",
-				"department_name": _("Sales"),
-				"parent_department": _("All Departments"),
-				"company": self.name,
+				"department_name": _("Sales")
 			},
 			{
-				"doctype": "Department",
-				"department_name": _("Purchase"),
-				"parent_department": _("All Departments"),
-				"company": self.name,
+				"department_name": _("Purchase")
 			},
 			{
-				"doctype": "Department",
-				"department_name": _("Operations"),
-				"parent_department": _("All Departments"),
-				"company": self.name,
+				"department_name": _("Operations")
 			},
 			{
-				"doctype": "Department",
-				"department_name": _("Production"),
-				"parent_department": _("All Departments"),
-				"company": self.name,
+				"department_name": _("Production")
 			},
 			{
-				"doctype": "Department",
-				"department_name": _("Dispatch"),
-				"parent_department": _("All Departments"),
-				"company": self.name,
+				"department_name": _("Dispatch")
 			},
 			{
-				"doctype": "Department",
-				"department_name": _("Customer Service"),
-				"parent_department": _("All Departments"),
-				"company": self.name,
+				"department_name": _("Customer Service")
 			},
 			{
-				"doctype": "Department",
-				"department_name": _("Human Resources"),
-				"parent_department": _("All Departments"),
-				"company": self.name,
+				"department_name": _("Human Resources")
 			},
 			{
-				"doctype": "Department",
-				"department_name": _("Management"),
-				"parent_department": _("All Departments"),
-				"company": self.name,
+				"department_name": _("Management")
 			},
 			{
-				"doctype": "Department",
-				"department_name": _("Quality Management"),
-				"parent_department": _("All Departments"),
-				"company": self.name,
+				"department_name": _("Quality Management")
 			},
 			{
-				"doctype": "Department",
-				"department_name": _("Research & Development"),
-				"parent_department": _("All Departments"),
-				"company": self.name,
+				"department_name": _("Research & Development")
 			},
 			{
-				"doctype": "Department",
-				"department_name": _("Legal"),
-				"parent_department": _("All Departments"),
-				"company": self.name,
+				"department_name": _("Legal")
 			},
 		]
-
-		# Make root department with NSM updation
-		make_records(records[:1])
-
-		frappe.local.flags.ignore_update_nsm = True
-		make_records(records)
-		frappe.local.flags.ignore_update_nsm = False
-		rebuild_tree("Department", "parent_department")
+		for dep_record in departments_list:
+			if not frappe.db.exists("Department", "{0} - {1}".format(dep_record["department_name"], self.abbr)):
+				department = frappe.get_doc(
+					{
+						"doctype": "Department",
+						"department_name": dep_record["department_name"],
+						"is_group": dep_record.get("is_group", 0),
+						"company": self.name,
+						"parent_department": "{0} - {1}".format(_("All Departments"), self.abbr) if not dep_record.get("is_group", 0) else "",
```

Comment:
All Departments won't have abbreviation at the end, all test cases failing because of it. Please fix and reopen

Reply:
I think abbreviations are required !
because each company has its own departments